src/input/cursor.c

```diff
@@ -1011,12 +1026,14 @@ cursor_button_release(struct seat *seat, uint32_t button,
 	seat_reset_pressed(seat);
 
 	if (server->input_mode == LAB_INPUT_STATE_MENU) {
-		if (ctx.type == LAB_SSD_MENU) {
-			menu_call_selected_actions(server);
-		} else {
-			menu_close_root(server);
-			cursor_update_common(server, &ctx, time_msec,
-				/*cursor_has_moved*/ false);
+		if (close_menu == 0 || time_msec - close_menu > rc.menu_hide_delay) {
```

Comment:
Could be `!close_menu` or we could just remove the condition here, `time_msec - 0` should always be `> rc.menu_hide_delay`.

Reply:
> I never verified if time_msec is actually the current time in ms, it *might* just be a serial.

I haven't either, but this worked fine in tests with x11, wayland, libinput+drm backends.